gradle-idea-configuration/src/main/groovy/com/palantir/gradle/ideaconfiguration/PluginDependency.java

```diff
@@ -16,36 +16,43 @@
 
 package com.palantir.gradle.ideaconfiguration;
 
-import java.io.Serializable;
-import java.util.Objects;
 import java.util.regex.Pattern;
 import java.util.stream.IntStream;
+import org.gradle.api.Named;
+import org.gradle.api.provider.Property;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.provider.SetProperty;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.Internal;
 
-public final class PluginDependency implements Serializable {
+public abstract class PluginDependency implements Named {
     private static final Pattern DOT_SPLITTER = Pattern.compile("\\.");
 
-    private final String name;
-    private String minVersion;
+    @Internal("Used to build up required min versions, not the final min version that is used")
+    public abstract SetProperty<String> getMinRequiredVersions();
```

Comment:
Currently, the existing API provides no way for a consumer of this plugin's API to lazily add required version, and if it was tweaked to do so, it would end up inspecting the current value each time and forcing it. Instead, we can just build up the required min versions and evaluate them all at the end.